### ISSUE TITLE
Feature Request: extract H1,H2,etc. tags as chapters

### ISSUE THREAD

**Issue by paul-chambers** [alexadam/save-as-ebook#49]
I'd like to be able to convert a page or selection as multiple chapters based on the location of H1, H2, etc tags, using the plain text between the tags as the chapter name.

As an example, I just converted the [Lua 5.3 Reference Manual](https://www.lua.org/manual/5.3/manual.html) to an eBook. It was a tedious process to manually select each chapter and capture it individually, then go back to edit the chapters to correct their titles. It would have been so much easier to just capture the page and have this extension be able to generate the chapters from the H1 and H2 tags on the captured page.

**Comment by alexadam** [comment]
a good idea but it will only work if the page is properly formatted with h1. h2... I can see a lot of issues from people trying to apply this to a 'modern' page that doesn't use <h>s but looks like one who does. I suppose you are dev and you can take a look at the source. Anyway, it can be disabled if the page does not contain enough <h>s

**Comment by paul-chambers** [comment]
Yes, I am an experienced dev, but on embedded Linux. I'm not fluent in JS, sadly.

I agree, if a page was authored purely with CSS style sheets and DIV tags, there's no simple way to recover the structure of the document. Having said that, many of the docs I want to convert, the formatting isn't that fancy, (or just CSS styles applied to the heading tags), so it would be useful to be able to recover it in one step. Luckily, it should be easy to spot the difference between a page that's using H1, H2 repeatedly vs. one that only uses them once or twice.
